src/compiler/transformers/declarations/emitResolver.ts

```diff
@@ -0,0 +1,397 @@
+                    }
+                    // If this is a function or method declaration, get the signature from the @type tag for the sake of optional parameters.
+                    // Exclude contextually-typed kinds because we already apply the @type tag to the context, plus applying it here to the initializer would supress checks that the two are compatible.
+                    result.push(decl);
+                }
+                return result;
+            }
+
+            if (nodeIsPresent((node as FunctionLikeDeclaration).body)) {
+                if (isGetAccessor(node) || isSetAccessor(node)) return false; // Get or set accessors can never be overload implementations, but can have up to 2 signatures
+                const symbol = getNodeLinks(node).symbol;
+                const signaturesOfSymbol = getSignaturesOfSymbol(symbol);
+                // If this function body corresponds to function with multiple signature, it is implementation of overload
+                // e.g.: function foo(a: string): string;
+                //       function foo(a: number): number;
+                //       function foo(a: any) { // This is implementation of the overloads
+                //           return a;
+                //       }
+                return signaturesOfSymbol.length > 1 ||
+                    // If there is single signature for the symbol, it is overload if that signature isn't coming from the node
+                    // e.g.: function foo(a: string): string;
+                    //       function foo(a: any) { // This is implementation of the overloads
+                    //           return a;
+                    //       }
+                    (signaturesOfSymbol.length === 1 && signaturesOfSymbol[0] !== node);
+            }
+            return false;
+        },
+        isOptionalParameter(parameter) {
+            const signature = parameter.parent;
+            const paramIndex = signature.parameters.indexOf(parameter);
+            Debug.assert(paramIndex !== -1);
+            if (parameter.questionToken) return true;
+            if (parameter.dotDotDotToken) return !!parameter.initializer;
+
+            for (let i = paramIndex; i < signature.parameters.length; i++) {
+                const p = signature.parameters[i];
+                if (!p.questionToken && !p.initializer && !p.dotDotDotToken) {
+                    return false;
+                }
+            }
+            return true;
+        },
+        isEntityNameVisible,
+        getTypeReferenceDirectivesForEntityName() {
+            return undefined;
+        },
+        isExpandoFunction,
+        getSymbolOfExternalModuleSpecifier() {
+            return undefined;
+        },
+        isImportRequiredByAugmentation() {
+            return false;
+        },
+    };
+
+    function isDeclarationVisible(node: Node): boolean {
+        if (node) {
+            const links = getNodeLinks(node);
+            if (links.isVisible === undefined) {

```

Comment:
This seems like it could just be `return links.isVisible ??= ...`, but I'm sure this is similar to some other code we have. (A lot of the code predates these operators.)

Reply:
This code was copied from the checker (mostly due to `getNodeLinks` not being accessible). I'll use the new operator.